src/Datadog.Trace/SpanContext.cs

```diff
@@ -73,7 +73,10 @@ internal SpanContext(SpanContext spanContext)
         /// </summary>
         public ulong SpanId { get; }
 
-        internal string ServiceName { get; set;  }
+        /// <summary>
```

Comment:
Does it make sense for a user to modify the `ServiceName` at the `SpanContext` level ? As far as I can tell, the only reason to access the `SpanContext` directly is to do cross-process propagation and in that case only the `TraceId` and `SpanId` are relevant.

Reply:
Makese sense. I made the `set` internal so it can be changed through a `Span` instance.

Reply:
My point stands for the get too, in my opinion it's only confusing for the user to see the `ServiceName` in the `SpanContext`. It is an implementation detail that we had to put it there, but in my opinion the only logical way to access it for a user is through the `Span` object.